### PR TITLE
Refactor migrate on schema utils

### PR DESCRIPTION
**Pull request by evilebottnawi**
**What kind of change does this PR introduce?**

refactor

**Did you add tests for your changes?**

existing tests

**Does this PR introduce a breaking change?**

Yes

- some plugins doesn't validate absolute paths, now it is validated, so configuration with relative paths in options can throw error
- `WatchIgnorePlugin` change constructor
- `ProfilingPlugin` accepts only absolute paths
- `webpack.validate` and `webpack.validateSchema` throw an error instead of returning an array or errors
- Validation Error name is now `ValidationError` instead of `WebpackOptionsValidationError`

**What needs to be documented once your changes are merged?**

Yes, see above

### CODE REVIEW

**Comment by evilebottnawi** [comment]
My mistake

**Comment by evilebottnawi** [comment]
Related issue https://github.com/webpack/webpack/issues/8503

File: schemas/plugins/debug/ProfilingPlugin.json [hunk 1]
```diff
@@ -6,8 +6,7 @@
     "outputPath": {
       "description": "Path to the output file e.g. `profiling/events.json`. Defaults to `events.json`.",
```
Comment (sokra) [inline schemas/plugins/debug/ProfilingPlugin.json]: Examples should showcase an absolute path. `path.resolve(__dirname, "profiling/events.json)`

**Pull request merged**